src/ResourceManager/SignalR/Commands.SignalR/Cmdlets/NewAzureRmSignalR.cs

```diff
@@ -0,0 +1,132 @@
+using Microsoft.Azure.Commands.SignalR.Models;
+using Microsoft.Azure.Commands.SignalR.Strategies;
+using Microsoft.Azure.Management.SignalR.Models;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Threading.Tasks;
+using Microsoft.Azure.Commands.SignalR.Strategies.ResourceManager;
+using System.Threading;
+using Microsoft.Azure.Commands.SignalR.Strategies.SignalRRp;
+
+namespace Microsoft.Azure.Commands.SignalR.Cmdlets
+{
+    [Cmdlet(VerbsCommon.New, SignalRNoun, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSSignalRResource))]
+    public sealed class NewAzureRmSignalR : SignalRCmdletBase
+    {
+        private const string DefaultSku = "Basic_DS2";
+
+        [Parameter(
+            Mandatory = false,
+            Position = 0,
+            HelpMessage = "The resource group name. The default one will be used if not specified.")]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            Position = 1,
+            HelpMessage = "The SignalR service name.")]
+        [ValidateNotNullOrEmpty()]
+        public string Name { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            Position = 2,
+            HelpMessage = "The SignalR service location. The resource group location will be used if not specified.")]
+        [LocationCompleter("Microsoft.SignalR/SignalR")]
+        [ValidateNotNullOrEmpty()]
+        public string Location { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The SignalR service SKU.")]
+        [PSArgumentCompleter("Basic_DS2")]
+        public string Sku { get; set; } = DefaultSku;
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The tags for the SignalR service.")]
+        public IDictionary<string, string> Tag { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Run the cmdlet in background job.")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+            => this.StartAndWait(SimpleExecuteCmdlet);
+
+        sealed class Parameters : IParameters<SignalRResource>
+        {
+            public string Location
+            {
+                get
+                {
+                    return _cmdlet.Location;
+                }
+
+                set
+                {
+                    _cmdlet.Location = value;
+                }
+            }
+
+            readonly NewAzureRmSignalR _cmdlet;
+
+            public Parameters(NewAzureRmSignalR cmdlet)
+            {
+                _cmdlet = cmdlet;
+            }
+
+            public async Task<ResourceConfig<SignalRResource>> CreateConfigAsync()
+            {
+                _cmdlet.ResourceGroupName = _cmdlet.ResourceGroupName ?? _cmdlet.Name;
```

Comment:
@ArieShout If there is a default resource group set, it will automatically be set as the value of ResourceGroupName.  Therefore, if ResourceGroupName is null, you know that the default value has not been set.